packages/atclient/src/atclient.c

```diff
@@ -259,7 +234,9 @@ int atclient_send_heartbeat(atclient *heartbeat_conn, bool listen_for_ack) {
   goto exit;
 
 exit: {
-  free(recv);
+  if (!heartbeat_conn->async_read) {
+    free(recv);
+  }
```

Comment:
I think we should avoid doing if checks on free-ing memory as much as possible to avoid reducing code complexity. free(NULL); is fine to do and recv is always = NULL unless pointed to memory location

Reply:
Because if we do this for a bunch of memory pointers, there'll be a gajillion if-checks and make it harder to read and track code